src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -0,0 +1,337 @@
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Responsible of "Configure > Advanced Parameters > Database -> SQL Manager" page
+ */
+class SqlManagerController extends FrameworkBundleAdminController
+{
+    /**
+     * Show list of saved SQL's
+     *
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/SqlManager/list.html.twig")
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    public function listAction(Request $request)
+    {
+        $searchForm = $this->createForm(FilterRequestSqlType::class, []);
+        $searchForm->handleRequest($request);
+
+        $filters = $this->get('prestashop.core.admin.search_parameters')->getFiltersFromRequest($request, [
+            'limit' => 10,
+            'offset' => 0,
+            'orderBy' => 'id_request_sql',
+            'sortOrder' => 'desc',
+            'filters' => $searchForm->getData(),
+        ]);
+
+        $repository = $this->getRepository();
+        $requestSqls = $repository->findByFilters($filters);
+        $requestSqlsCount = $repository->getCount();
+
+        $settingsForm = $this->getSettingsFormHandler()->getForm();
+
+        $data = [
+            'request_sqls' => $requestSqls,
+            'request_sqls_count' => $requestSqlsCount,
+            'order_by' => $filters['orderBy'],
+            'order_way' => $filters['sortOrder'],
+            'filters' => $filters,
+            'search_form' => $searchForm->createView(),
+            'settings_form' => $settingsForm->createView(),
+        ];
+
+        return $this->getTemplateParams($request) + $data;
+    }
+
+    /**
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/SqlManager/table.html.twig")
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    public function tableAction(Request $request)
+    {
+        $tableFactory = $this->get('prestashop.core.table.factory');
+        $tableViewFactory = $this->get('prestashop.core.table.view_factory');
+        $tableDefinitionFactory = $this->get('prestashop.core.table.factory.request_sql_definition');
+        $tableDataProvider = $this->get('prestashop.core.table.data_provider.request_sql');
+
+        $tableDefinition = $tableDefinitionFactory->createNew();
+
+        $table = $tableFactory->createFromDefinition($tableDefinition, $request);
+
+        $filters = [
+            'limit' => $request->query->get('limit', 10),
+            'offset' => $request->query->get('offset', 0),
+            'orderBy' => $request->query->get('orderBy', $tableDefinition->getDefaultOrderBy()),
+            'sortOrder' => $request->query->get('sortOrder', $tableDefinition->getDefaultOrderWay()),
+            'filters' => $table->getForm()->getData(),
+        ];
+
+        $table->setRows($tableDataProvider->getRows($filters));
+        $table->setRowsTotal($tableDataProvider->getRowsTotal());
+
+        return [
+            'table' => $tableViewFactory->createViewFromTable($table),
```

Comment:
we could make $tableViewFactory available in FrameworkAdminBundle and do something like

```php
'table' => $this->createTableView($table),
```